src/docs/reference/modules/Shortcodes/README.md

```diff
@@ -1,11 +1,137 @@
+
+### Example Shortcode Templates :
+
+#### `[display_text]`
+
+| Parameter | Value |
+| --------- | ----------- |
+| `Name` | [display_text] |
+| `Hint` | Returns the display text of the content item. |
+| `Usage` | [display_text] |
+| `Content` | `{{ Context.ContentItem.DisplayText }}`<br>`{{ More}}` |
+
+!!! note
+    The `ContentItem` `Context` is only available when the caller, i.e. an `HtmlBodyPart`, has passed the `ContentItem` value to the `Context`. 
+
+#### `[site_name]`
+
+| Parameter | Value |
+| --------- | ----------- |
+| `Name` | [site_name] |
+| `Hint` | Returns the site name. |
+| `Usage` | [site_name] |
+| `Content` | `{{ Site.SiteName }}` |
+
+####  `[primary]`
+
+| Parameter | Value |
+| --------- | ----------- |
+| `Name` | [primary] |
+| `Hint` | Formats text in the themes primary color. |
+| `Usage` | [primary text]&lt;br&lt;[primary]text[/primary] |
+| `Content` | `{% capture output %}`<br>`{% if Args.Text != nil %}`<br>`<span class="text-primary">{{Args.Text}}</span>`<br>`{% else %}`<br>`<span class="text-primary">{{Content}}</span>`<br>`{% endif %}`<br>`{% endcapture %}`<br>`{{ output | sanitize | raw }}` |
+
+## Shortcodes via code
+
+### Shortcode Delegate
+
+Shortcodes can be registered in code via a `ShortcodeDelegate` using the `AddShortcode` extension method.
+
+| Parameter | Type | Description |
+| --------- | ---- |------------ |
+| `Name` | `string` | The name of the Shortcode. |
+| `Shortcode` | `ShortcodeDelegate` | The Shortcode Delegate. |
+| `Describe` | `Action<ShortcodeOption>` | Optionally a Description of the Shortcode. |
+
+In this example we register a `[bold]` Shortcode Delegate and describe the Shortcode.
+
+``` csharp
+services.AddShortcode("bold", (args, content, ctx) => {
+    var text = args.NamedOrDefault("text");
+    if (!String.IsNullOrEmpty(text))
+    {
+        content = text;
+    }
+
+    return new ValueTask<string>($"<b>{content}</b>");
+}, describe => {
+    describe.ReturnShortcode = "[bold text-here]";
+    describe.Hint = (serviceProvider) => {
+        var S = serviceProvider.GetRequiredService<IStringLocalizer<Startup>>();
+        return S["Add bold formatting with a shortcode."];
+    };
+    describe.Usage = "[bold 'your bold content here']";
+    describe.Categories = (serviceProvider) => {
+        var S = serviceProvider.GetRequiredService<IStringLocalizer<Startup>>();
+        return new string[] { S["HTML Content"] };
+    };
+});
+```
+
+### `IShortcodeProvider`
+
+Shortcodes may also be used by implementing the `IShortcodeProvider` interface and registered with the `AddShortcode` extension method.
+
+In this example we register an `ImageShortcodeProvider` as `[image]` and describe the Shortcode.
+
+``` csharp
+services.AddShortcode<ImageShortcodeProvider>("image", describe => {
+    describe.ReturnShortcode = "[image] [/image]";
+    describe.Hint = (serviceProvider) => {
```

Comment:
No localization here, will use dynamic localization when available. Even though this is registered by code. We are following what we do with content types.